ast/compile.go

```diff
@@ -1207,8 +1207,14 @@ func (vis *ruleArgLocalRewriter) Visit(x interface{}) Visitor {
 		// Scalars are no-ops. Comprehensions are handled above. Sets must not
 		// contain variables.
 		return nil
+	case Call:
+		// Don't rewrite the call name
```

Comment:
Nit: this comment would be more helpful if it explained why we're not rewriting the call head.

It would be useful to enumerate/test the following cases:

* Head refers to a global (e.g., `f(g(1)) { ... }` where g is another user-defined function (`g` should be resolved to `data.x.g` though so maybe it's not interesting unless body declares a var called `data`.)
* Head refers to a local variable (e.g., `f(g(1)) { g := {"foo": "bar"}; ... }`)
* Head refers to a built-in function where the var is declared in the function (e.g., `f(object.get(1)) { object := {"a": 1} }`)

Reply:
👍 good call on these test cases.. they turned up some issues with the local vars that have conflicting names overwriting them in the body 😞 Will fix and update the PR